### PR TITLE
Changelings

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR
Added changelings, based on [Paradise Station wiki entry](https://paradisestation.org/wiki/index.php/Changeling) with a bit of [this design doc](https://github.com/space-wizards/docs/pull/263). They have their differences but they're mostly the same.
With custom objectives to make them somewhat different from a tot.

Objectives:
- **Survive.** Pretty obvious, since your biomass is running out.
- **Extract 6-9 genomes.** You can either absorb someone to get their DNA or use the stealth sting ability.
- **Escape on the shuttle while being "person".** Simply transform into that person and escape on the shuttle.

Some can say they're entirely based on round removal.

Absorption gives you bonus evolution points and increases the maximum amount of chemicals. Absorbing another changeling gives you half of it's max chemicals, biomass and evolution points.

## Why / Balance
:trollface: 
image
EDIT: thank god someone made an actually approved and merged design doc for clings :tm:

## Technical details
shitcode!! need 3000 refactors and 1984 comments

## Media
Since i for some reason can't build it because of some git pull robust toolbox clustershit I'm showing media taken from Goobstation, where changelings were playtested before PRing it to wizden.

Basic abilities, objectives, blabla
if icons ever tell you anything
image
image
image
image

Gaming setup (space adaptation + armblade + shield)
image
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
none i believe.

## TODO
- [X] Wait for changeling's design doc to be passed. Holy shit
- [ ] Playtest it
- [ ] Listen to other people's opinions

NEW TODO:
- [ ] Listen to even more opinions
- [ ] Do bandaid fixes
- [X] Add a ling check

NEW NEW TODO:
- [ ] Refactor ability system because it's dogshit at the moment

All of the mechanics are fully implemented so this can be copied/merged and happily used.
Downstream maintainers go crazy ❤️😔

**Changelog**
:cl:
- add: The changeling hive has boarded the station, ready to take anything it desires - be it your faces, equipment or your lives!

### CODE REVIEW

**Comment by whateverusername0** [comment]
forgot to port the gamerule my bad og

**Comment by whateverusername0** [comment]
@ARMOKS you can pull up w/ the design doc here ❤️ :trollface:

**Comment by Verbalase** [comment]
finally

peak

**Comment by whateverusername0** [comment]
> Hooray, I hope you fixed the slow chem regen.

hope dies last :trollface:

**Comment by Djungelskog2** [comment]
armok made a doc? can someone reply me the pr?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
1. Please read through https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html?highlight=conventions#conventions
2. Almost all of this code is hardcoded to changelings and not generic.
3. While goobstation may not care about asset licensing we do.

**Comment by whateverusername0** [comment]
> 1. Please read through https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html?highlight=conventions#conventions

that's great but where does my code start being non-conventional?

> 2. Almost all of this code is hardcoded to changelings and not generic.

it makes sense because changelings are kinda not supposed to be generic, because every ability requires chemicals and if you're not a changeling you're not supposed to use them, exception being an omnitraitor but it's easily mitigatable by simply adding a changeling component. methods are also are made as generic as possible i can't go any farther beyond making a million methods to do a single thing, it's just dumb imo.

> 3. While goobstation may not care about asset licensing we do.

my bad og, forgot to reference a couple things, i'll do it tomorrow or sth

**Comment by Djungelskog2** [comment]
hope dies last 😔

**Comment by FaDeOkno** [comment]
Can you make a radial menu for transformation? 
It's possible - you can save info about saved DNAs like markings, species and other, build each form on the client and use SpriteView to set your result as button sprite
Anyway nice work

**Comment by DVOniksWyvern** [comment]
Goobcode is superior to wizden code, it seems, can't have that here. Think about the admins, who might get confused by the shapeshifting!

**Comment by EmoGarbage404** [comment]
@DVOniksWyvern if you want to jerk about your favorite server please do so elsewhere.

**Comment by whateverusername0** [comment]
> Can you make a radial menu for transformation? It's possible - you can save info about saved DNAs like markings, species and other, build each form on the client and use SpriteView to set your result as button sprite Anyway nice work

sure, i'll try 👍

**Comment by FaDeOkno** [comment]
Good luck with that

**Comment by Verbalase** [comment]
we arent getting lings

sadge



> Goobcode is superior to wizden code, it seems, can't have that here. Think about the admins, who might get confused by the shapeshifting!

oniks you are cool and all, but idk if taking that here would be good. all you though gng.

**Comment by idoffront** [comment]
нихуева

**Comment by DVOniksWyvern** [comment]
> apologies for having standards. we should clearly just allow whatever code people submit because it's "cool." That has clearly never caused issues before in the past (disease, vehicles, executions, nukies)

Aren't you marked as reviewer on this PR? Then do it - review the code, tell the man, what exactly should be changed and how lines of code should be so aforementioned "issues" wouldn't be a stopper here. Whatever you are doing is a sad affair by now, you do not have anything that is considered "good", "cool" or "big". So yes, for the sake of pure competitiveness - it's me who should be telling you to "pull your head out from your own ass".

> keep working at it.

Am I wrong in my observation? What exactly you've personally done in the past six month and that wasn't PRed by outside contributor? 

> oniks you are cool and all, but idk if taking that here would be good. 

My G, there's always time for that. I am self-reflecting.

**Comment by Verbalase** [comment]
> > apologies for having standards. we should clearly just allow whatever code people submit because it's "cool." That has clearly never caused issues before in the past (disease, vehicles, executions, nukies)
> 
> Aren't you marked as reviewer on this PR? Then do it - review the code, tell the man, what exactly should be changed and how lines of code should be so aforementioned "issues" wouldn't be a stopper here. Whatever you are doing is a sad affair by now, you do not have anything that is considered "good", "cool" or "big". So yes, for the sake of pure competitiveness - it's me who should be telling you to "pull your head out from your own ass".
> 
> > keep working at it.
> 
> Am I wrong in my observation? What exactly you've personally done in the past six month and that wasn't PRed by outside contributor?
> 
> > oniks you are cool and all, but idk if taking that here would be good.
> 
> My G, there's always time for that. I am self-reflecting.

alrighty, ye be the boss

**Comment by Ilya246** [comment]
"can't you just do <thing that may take a long while to complete> in 5 seconds the instant it becomes possible"
if you want to have criticisms i suggest learning what you're talking about first so they have more weight

**Comment by beck-thompson** [comment]
> Aren't you marked as reviewer on this PR? Then do it - review the code, tell the man, what exactly should be changed and how lines of code should be so aforementioned "issues" wouldn't be a stopper here.

Emo and most of the other maintainers are very helpful if you just ask for help / clarification on discord! There is almost 350 open pull requests right now so they are all busy. I guess I'm just saying you might want to cut them some slack 😆! Code reviews are hard and take time to do correctly.

Side note: Love the PR! Good luck @whateverusername0🫡

**Comment by Djungelskog2** [comment]
dont forget that they are all volunteers and even if it doesnt get accepted, chances are someone or somewhere else is

**Comment by deathride58** [comment]
The code here is a pretty straight downgrade of changeling code as present across the SS13 servers that have them. Even in SS13, changeling abilities are split off into their own objects (the equivalent here would be components+systems), regardless of what server you're looking at (even Goon has it that way, despite it hailing from an era where Goon's standards of code quality were pretty different from the standards across the rest of SS13).

That aside, these are also the types of lings that the maintainer team has previously agreed in the past to be outright counterproductive to the game's health; these are the brand of ling that're ultimately based off the game Prototype, which are rather infamous for being outright overpowered as a consequence of years worth of powercreep. The original horror themeing of lings is more or less completely lost (and certain things that lean heavily into that horror aspect, such as true form, are simply gone outside of Baycode (where antags rarely see the sun outside of admin events), and Goonstation (where lings instead go hard towards being modeled after The Thing; the complete opposite direction from where most SS13 servers went)), in favor of lings becoming incredibly combat-oriented (it's genuinely quite rare to see ling players actually use their stealth-oriented tools on servers that feature Prototype lings, as the tools they have for going loud are simply overpowering to the point where they don't have to care about stealth at all).

This kind of nuance is the reason why antags, such as changelings, require a design doc (one that's not just a copy-paste or paraphrase of an SS13 server's wiki) in order to be merged; SS13 is simply *not* an ideal role model when it comes to design, and for things like changelings, there's a *lot* of room for genuine creativity that simply isn't seen on most SS13 servers. Ideally, an implementation of changelings should strive to be far more interesting from a design standpoint than what they're like on most SS13 servers today.

**Comment by Djungelskog2** [comment]
I think loads of people will agree with me though, even if they are content with the game no being 1:1 they came ss14 for a version of 13 thats easier to play, including most major stuff like antags. I dont see the issue with almost copying specific variations of antags from 13 if its balanced and holds up well (with maintenance of course). Also from the points you brought up outside of the 'Be creative/Original'  sounds like its just asking for an implementation of baystation or goon lings over TG, am i misunderstanding something?

**Comment by deathride58** [comment]
> I dont see the issue with almost copying specific variations of antags from 13 if its balanced

Prototype lings simply aren't balanced; almost every ability they have is a straight upgrade of the tools available to traitors, with very few lacking traitor equivalents. They are, for all intents and purposes, just traitor++. It's incredibly common to see lings end up being capable of taking on the entire station solo, even on highpop servers.

> 'Be creative/Original' sounds like its just asking for an implementation of baystation or goon lings over TG, am i misunderstanding something?

The reason Bay and Goon's ling implementations are brought up is because they serve as examples of how changelings used to be across SS13. Most notably, how they used to lean heavily towards being a proper space horror, and in turn, quite a bit more interesting from a gameplay and design standpoint. However, that horror aspect is something that's been held back by a combination of SS13's tech debt, Byond's limitations at the time of design, and a general lack of further design iteration.

**Comment by Djungelskog2** [comment]
> > I dont see the issue with almost copying specific variations of antags from 13 if its balanced
> 
> Prototype lings simply aren't balanced; almost every ability they have is a straight upgrade of the tools available to traitors, with very few lacking traitor equivalents. They are, for all intents and purposes, just traitor++. It's incredibly common to see lings end up being capable of taking on the entire station solo, even on highpop servers.
> 
> > 'Be creative/Original' sounds like its just asking for an implementation of baystation or goon lings over TG, am i misunderstanding something?
> 
> The reason Bay and Goon's ling implementations are brought up is because they serve as examples of how changelings used to be across SS13. Most notably, how they used to lean heavily towards being a proper space horror, and in turn, quite a bit more interesting from a gameplay and design standpoint. However, that horror aspect is something that's been held back by a combination of SS13's tech debt, Byond's limitations at the time of design, and a general lack of further design iteration.

ok yea i think im just not understanding, I agree that TG and TG-like lings are basically Traitor+, but im unsure wether you think goon lings should be something to strive for, with rebalancing to appropriate it for the version of space station its entering of course or if you think something else should be done altogether

**Comment by deathride58** [comment]
> im unsure wether you think goon lings should be something to strive for, (...) or if you think something else should be done altogether

Anything that ain't Prototype lings would be ideal (or at least, anything resembling their implementation across SS13). Goon lings would indeed make for a pretty good base for ling design, but the design of Goon lings come from a time when The Thing was more or less the only media inspiration for them. Since the original implementation of changelings, there's been a pretty wide variety of further thematically-appropriate media published that expand upon the core concepts, such as White Wolf's Changeling: The Lost (which leans even heavier towards horror, even for the titular changelings themselves), Prey 2017's typhon (which leans a bit more towards action, but without the one-man-army aspect), and more (if we rattled off every piece of scifi or horror media with shapeshifters, we'd be here all day).

Where the inspiration comes from, and generally what direction to take lings in, is ultimately up to whoever writes the design doc. A designer who knows what they're doing could potentially end up making Prototype-inspired lings work well from a design standpoint (which would require actively acknowledging the faults of SS13's current implementations of Prototype lings, and somehow managing to address that). However, the proper place to bikeshed this would be in either the Discord's design channels, or a design doc PR.

**Comment by wafehling** [comment]
Is there a guide/documentation on how the antag refactor works? It'd go a long way towards adding new antags/gamemodes/events in general I feel, something we could definitely use.

**Comment by Agustron24** [comment]
If we dont like it begin Paradise, we can rework it later.

Also holy shit why did shit start raining.

**Comment by EmoGarbage404** [comment]
> If we dont like it begin Paradise, we can rework it later.

there's no point in implementing an antagonist if we're going to have to toss chunks of it due to design issues that we can catch early.

**Comment by metalgearsloth** [comment]
> > 1. Please read through [docs.spacestation14.com/en/general-development/codebase-info/conventions.html?highlight=conventions#conventions](https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html?highlight=conventions#conventions)
> 
> that's great but where does my code start being non-conventional?

1. Datafield strings have been obsolete for a while.
2. Assets require licensing / proper licensing.
3. You hardcoded in every single thing to changeling so if anyone wants to modify it they need to touch the code heavily and none of it can be re-used to make more antags in future, which emo just spent a great deal of effort fixing. ChangelingComponent + ChangelingSystem in particular are unusable. For one example you hardcode in reagents to "AnatomicPanacea" with 0 tests which will just silently break in future. There's many more instances of this in the file.
4. None of it serializes properly with datafields and will break with persistence.
5. Most if it isn't documented.
6. Datafields with VVRW are redundant.
7. Objectives & categories are hardcoded into changeling rule.
8. Much of ChangelingRule looks like it was copy-pasted from an obsolete system. If you have to copy code from somewhere you should stop and ask if it's really necessary which in this case it isn't.
9. ChangelingSystem copies so much existing code that we already have. You even commented it's "yanderedev type shit" when the entire system needs to be redone.
10. You mix fields and methods together which makes it a massive pain.
10. None of it's predicted. We've been asked for all new code to be predicted for a while as well.

Additionally we've said before we would prefer not to have the newer tg-style lings that are just powergamers and would prefer something more akin to goon.

**Comment by whateverusername0** [comment]
ok thanks now i've got a solid idea of what to do

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:524]: You can use QueueDel(EntityUid) instead of this long lines

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:629]: Changelings should be able to refresh their store and lose all abilities to select other, not just farm evo points by commiting mass genocide

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:740]: You can use RejuvenateEvent (but it will remove handcuffs too)

**Comment by SlamBamActionman** [comment]
> ok thanks now i've got a solid idea of what to do

Write a design document?

Okay I'm being a bit facetious, but really it should be submitted as a design document PR first, code PR second.

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:876]: Isn't it a bit too much? Max eye health is 9

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:1010]: Not removing StealthOnMove can cause bugs as I remember

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:1098]: Shouldn't it be a passive effect that activates right after bying?

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:771]: Use Spawn(EntityUid, Coordinates) instead of this

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (FaDeOkno) [inline Content.Server/Changeling/ChangelingSystem.cs:479]: Do the same thing with MindShield comp. Lings can't copy it, but also can't just destroy

**Review by FaDeOkno — COMMENTED** [review]
Why do you use EntityManager in EntitySystem?

**Comment by whateverusername0** [comment]
> Why do you use EntityManager in EntitySystem?

forgot about that

**Comment by Kukutis96513** [comment]
attempted to playtest this but uhh, i can't turn myself into a changeling via the admin menu so i had to add the changeling component myself, found a few issues though:
1. The evolution menu doesn't seem to work?
2. I can't absorb anyone.
3. Transforming causes me to drop my PDA.
4. You can't transform back to yourself.

**Comment by hord-brayden** [comment]
@whateverusername0 fwiw - this is some fine work! Hats off to you, I've been reading through this for a bit, it's interesting and very exciting, great job!

**Comment by lanse12** [comment]
@Kukutis96513 works perfectly on goob, that's an issue on your part

**Comment by deathride58** [comment]
Neither 3 nor 4 are present in any SS13 server featuring changelings. In SS13, you're given a DNA copy of your character at roundstart as ling (as the entire core concept of changeling is that a shapeshifting alien has snuck aboard by taking the place of your character). 3 is strongly likely to be a bug given that the polymorph is explicitly set to transfer the entire inventory over (which *should* be including the ID slot)

Also, Goobstation is not Wizden; code ripped straight from downstreams are not guaranteed to work out of the box due to the wide variety of changes that can be present on downstreams. It's up to the PR author's responsibility to ensure their code works in the environment they've ported it to.

**Comment by whateverusername0** [comment]
my bad og

**Comment by Dutch-VanDerLinde** [comment]
happens because when transfering inventory, the jumpsuit slot is transferred first, which causes the PDA to drop as the PDA depends on the jumpsuit

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (whateverusername0) [reply Content.Server/Changeling/ChangelingSystem.cs:1098]: iunno how to make such 😅

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (whateverusername0) [reply Content.Server/Changeling/ChangelingSystem.cs:876]: didn't know

**Comment by whateverusername0** [comment]
i'm working on changelings on goob station right now and simply porting my changes here, so don't expect frequent updates as it needs to get playtested beforehand 😅

**Comment by a-person5660** [comment]
Great to hear that this is being worked on actively!

**Comment by RedNiloc** [comment]
Thank you for actively working on this, hyped to finally get ling after so long.

**Comment by whateverusername0** [comment]
I have come back once more to reming you that the changeling is real.
Did a ton of bug fixes and other additions, finally porting it here.

**Comment by sad2284** [comment]
Please accept.. Anyway, thank you.

**Comment by a-person5660** [comment]
Don't forget to update the pr description bit with the new way that the ling works.

**Comment by chavonadelal** [comment]
Maintainers please help the author with the pull request so that it can finally be implemented in ss14. And don't leave him 1 on 1 with problems if you don't like something.

**Comment by whateverusername0** [comment]
> Maintainers please help the author with the pull request so that it can finally be implemented in ss14. And don't leave him 1 on 1 with problems if you don't like something.

but nobody came. 🤷‍♂️

**Comment by whateverusername0** [comment]
i guess nobody is going to review it and it's just gonna stay there.
oh well, i guess it's on the most visible spot anyways

**Comment by metalgearsloth** [comment]
Stop complaining about 4-day old changes not being reviewed please.

**Comment by whateverusername0** [comment]
> Stop complaining about 4-day old changes not being reviewed please.

you say that like you singlehandedly reviewed the entire thing

**Comment by metalgearsloth** [comment]
> > Stop complaining about 4-day old changes not being reviewed please.
> 
> you say that like you singlehandedly reviewed the entire thing

I reviewed the initial PR.

You made changes 4 days ago and are whining about it not having been checked. There are other contributors that also need their PRs to be reviewed so be considerate of others.

**Comment by whateverusername0** [comment]
where do you see me complaining mane, it's all acceptance

**Comment by RedNiloc** [comment]
Can head slug / lesser form actually vent crawl?

**Comment by whateverusername0** [comment]
> Can head slug / lesser form actually vent crawl?

no and i believe we'll never get that feat

**Comment by VigersRay** [comment]
> > Can head slug / lesser form actually vent crawl?
> 
> no and i believe we'll never get that feat

https://github.com/space-wizards/space-station-14/pull/18182

File: Content.Server/Changeling/ChangelingSystem.cs
Comment (VigersRay) [inline Content.Server/Changeling/ChangelingSystem.cs:364]: I will be very glad if you make a separate system for this that can be used not only for changeling.
At the very least, I need it for the cult of flesh and also for the IPC charging module.

**Comment by whateverusername0** [comment]
forgot to port client side changes, now biomass and chemicals should render properly 😅

**Comment by whateverusername0** [comment]
a new batch of bugs got spammed into my DMs (again), expect update rolling out soon enough
also planning on refactor the whole thing because i agree that hardcoded reagents are bad (and other stuff too)
but that's gonna come in like a week™️ :trollface:

**Comment by sad2284** [comment]
> Новая партия ошибок была заспамлена в моих личных сообщениях (снова), ожидайте обновления достаточно скоро, также планирую рефакторинг всего этого, потому что я согласен, что жестко запрограммированные реагенты плохи (и другие вещи тоже), но это произойдет примерно через неделю™️ :троллфейс:
> 
> Также вышел новый документ по дизайну, я могу воплотить его в жизнь 😄, просто я не хочу прощаться со всеми способностями, которые уже есть. Предложения?

Please bring your case to the end. I want this antagonist to appear in our favorite game

**Comment by RedNiloc** [comment]
> a new batch of bugs got spammed into my DMs (again), expect update rolling out soon enough also planning on refactor the whole thing because i agree that hardcoded reagents are bad (and other stuff too) but that's gonna come in like a week™️ :trollface:
> 
> also new design doc is out, i can make it real 😄, it's just that i don't want to say byebye to all the abilities that are there already. suggestions?

when you mean "I don't want to say byebye to all the abilities" does that mean they are going to be removed or reworked?

**Comment by whateverusername0** [comment]
> when you say "I don't want to say byebye to all the abilities" does that mean they are going to be removed or reworked?

removed to match the design doc

**Comment by RedNiloc** [comment]
How closely do you have to follow the design doc? Because the doc mentions that virology would have to be added before changelings due to the eye spiders inflicting a special disease. I was wondering if you could just use chems instead (for example ambuzol stopping zombie virus), That way you are not waiting on a update to add viro and you can just get changelings out the door.

**Comment by whateverusername0** [comment]
> How closely do you have to follow the design doc?

good question. that's why i'm asking if i'll have to remove most of ling's abilities that already exist

**Comment by AverageNotDoingAnythingEnjoyer** [comment]
so uh, how's changelings pr, what now?

**Comment by RedNiloc** [comment]
For now, we wait until he adds new stuff (hopefully) or reviewers get around to checking the PR.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Needs predicting.

After having coded AI I have a better idea of what needs doing. For starters all of the stings being hardcoded reagents is bad. These should just be components stored on the ling itself or on an action entity.

Another if you have actions applying component changes when we already have functionality to do this entirely in yaml via compregistry.

Once you've done the above we'll go from there.

**Comment by whateverusername0** [comment]
> Another if you have actions applying component changes when we already have functionality to do this entirely in yaml via compregistry.

ok cool. where can i find an example of that so that i can work my way up from there?

**Comment by whateverusername0** [comment]
btw while i'm working on heretics this repo will be put on hold

**Comment by TokenStyle** [comment]
> ok cool. where can i find an example of that so that i can work my way up from there?

compregistry = `ComponentRegistry`, here's examples:
AntagSelectionComponent.cs, DataDefinition https://github.com/space-wizards/space-station-14/blob/master/Content.Server/Antag/Components/AntagSelectionComponent.cs#L144
NinjaAntag .yml as example using AntagSelectionComponent, DataDefinitions. Here also components have unique data in components https://github.com/space-wizards/space-station-14/blob/master/Resources/Prototypes/GameRules/events.yml#L199
AntagSelectionSystem https://github.com/space-wizards/space-station-14/blob/master/Content.Server/Antag/AntagSelectionSystem.cs#L162

As you can see, these components from `ComponentRegistry` will be aplied with all their changes in yml "as is" in `AntagSelectionSystem`, so you don't need to hard-code in c# classes or in Components, for example, you can make `public ComponentRegistry CompsToAddOnStartup = new();` and all these components would applied (with some logic in System's class of course) to the entity with custom settings in CompsToAddOnStartup list/registry.

**Comment by AverageNotDoingAnythingEnjoyer** [comment]
are you going to keep the hunger system? your pr has been merged in one of the servers and from my experience it results in changelings slaughtering a lot of people, as every single changeling has to consume someone every 20-25 minutes

**Comment by whateverusername0** [comment]
> are you going to keep the hunger system? your pr has been merged in one of the servers and from my experience it results in

it's in the design doc so yes
perhaps we can add some more mutagen effects to simplemobs like turning monkeys into catatonic humans

**Comment by RedNiloc** [comment]
> are you going to keep the hunger system? your pr has been merged in one of the servers and from my experience it results in changelings slaughtering a lot of people, as every single changeling has to consume someone every 20-25 minutes

I feel like it should be every 26 mins or so, but I feel like the point of a changeling is to consume alot and to be a very active and more threatening then syndicates, that way it puts pressure on the crew and on security to catch them.

**Comment by whateverusername0** [comment]
can't bring myself to refactor it, heck can't even bring myself to actually do any coding. feel free to use this pr in any way you want, i'm out.

**Comment by beck-thompson** [comment]
> can't bring myself to refactor it, heck can't even bring myself to actually do any coding. feel free to use this pr in any way you want, i'm out.

I feel that ❤️

**Pull request closed without merging**
